sys/linux/security_ima.txt

```diff
@@ -0,0 +1,111 @@
+openat$ima_violations(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/violations"]], flags const[O_RDONLY], mode const[0]) fd_ima_count
+read$ima_count(fd fd_ima_count, buf ptr[out, fmt[dec, intptr]], count len[buf])
+
+ima_condition_elem [
+# base
+	func		fs_opt["func", stringnoz[ima_funcs]]
+	mask		fs_opt["mask", stringnoz[ima_masks]]
+	fsmagic		fs_opt_hex["fsmagic", intptr]
+	fsuuid		fs_opt["fsuuid", uuid_str]
+	uid_eq		fs_opt_dec["uid", uid]
+	euid_eq		fs_opt_dec["euid", uid]
+	fowner_eq	fs_opt_dec["fowner", uid]
+	uid_gt		fs_opt_nodelim["uid>", fmt[dec, uid]]
+	euid_gt		fs_opt_nodelim["euid>", fmt[dec, uid]]
+	fowner_gt	fs_opt_nodelim["fowner>", fmt[dec, uid]]
+	uid_lt		fs_opt_nodelim["uid<", fmt[dec, uid]]
+	euid_lt		fs_opt_nodelim["euid<", fmt[dec, uid]]
+	fowner_lt	fs_opt_nodelim["fowner<", fmt[dec, uid]]
+# lsm
+	obj_user	fs_opt_str["obj_user"]
+	obj_role	fs_opt_str["obj_role"]
+	obj_type	fs_opt_str["obj_type"]
+	subj_user	fs_opt_str["subj_user"]
+	subj_role	fs_opt_str["subj_role"]
+	subj_type	fs_opt_str["subj_type"]
+# option
+	appraise_type	fs_opt["appraise_type", stringnoz[ima_appraise_type]]
+	appraise_flag	stringnoz["appraise_flag=check_blacklist"]
+	permit_directio	stringnoz["permit_directio"]
+	keyrings	fs_opt["keyrings", ima_keyring_list]
+	pcr		fs_opt_dec["pcr", int64[0:64]]
+] [varlen]
+
+ima_condition {
+	cond	ima_condition_elem
+	sp0	const[' ', int8]
+} [packed]
+
+ima_policy_line {
+	action	stringnoz[ima_action]
+	sp0	const[' ', int8]
+	list	array[ima_condition]
+	n	const[0xa, int8]
+} [packed]
+
+ima_policy_list {
+	list	array[ima_policy_line]
+	z	const[0, int8]
+} [packed]
+
+ima_policy [
+	file	filename
+	policy	ima_policy_list
+] [varlen]
+
+resource fd_ima_policy[fd]
+openat$ima_policy(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/policy"]], flags const[O_RDWR], mode const[0]) fd_ima_policy
+write$ima_policy(fd fd_ima_policy, buf ptr[in, ima_policy], count len[buf])
+read$ima_policy(fd fd_ima_policy, buf ptr[out, ima_policy], count len[buf])
+
+evm_flags = EVM_INIT_HMAC, EVM_INIT_X509, EVM_ALLOW_METADATA_WRITES, EVM_SETUP_COMPLETE
+
+resource fd_evm[fd]
+openat$evm(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/evm/evm"]], flags const[O_RDWR], mode const[0]) fd_evm
+write$evm(fd fd_evm, buf ptr[in, fmt[dec, flags[evm_flags]]], count len[buf])
+read$evm(fd fd_evm, buf ptr[out, fmt[dec, flags[evm_flags]]], count len[buf])
+
+xattr_name_random_noz {
+	prefix	stringnoz[xattr_prefix]
+	name	stringnoz
+} [packed]
+
+evm_xattr_name [
+	known	stringnoz[xattr_names]
+	random	xattr_name_random_noz
+	point	const['.', int8]
+] [varlen]
+
+evm_xattr_line {
+	xattr	evm_xattr_name
+	n	const[0xa, int8]
+} [packed]
+
+evm_xattr_list {
+	list	array[evm_xattr_line]
+	z	const[0, int8]
```

Comment:
Not that it matters much, but I wonder if 0 is really required. Generally kernel code should look at the size of the write operation and 0-terminate itself if it needs the 0.